lib/solidus_klarna_payments/engine.rb

```diff
@@ -17,11 +17,21 @@ class Engine < Rails::Engine
     end
 
     initializer 'spree.solidus_klarna_payments.payment_methods', after: 'spree.register.payment_methods' do |app|
-      app.config.spree.payment_methods << Spree::PaymentMethod::KlarnaCredit
+      app.config.to_prepare do
+        app.config.spree.payment_methods << ::Spree::PaymentMethod::KlarnaCredit
+      end
     end
 
     config.to_prepare do
       ::Spree::PermittedAttributes.source_attributes << :authorization_token
     end
+
+    initializer 'solidus_klarna_payments.pub_sub' do |app|
+      unless SolidusSupport::LegacyEventCompat.using_legacy?
+        app.reloader.to_prepare do
```

Comment:
why `reloader` and not `config` here?

Reply:
Good point. It seems that [both do the same under the hood](https://github.com/rails/rails/blob/fe76a95b0d252a2d7c25e69498b720c96b243ea2/railties/lib/rails/application/finisher.rb#L104-L108). Using `app.reloader.to_prepare` everywhere for consistency.